src/index.js

```diff
@@ -285,63 +286,47 @@ const proxyCompare = (a, b, locations) => {
   return ret;
 };
 
-const nestedSort = (a, b) => {
-  for (let i = 0; ; ++i) {
-    const ac = a.charCodeAt(i);
-    const bc = b.charCodeAt(i);
-    if (!ac && !bc) return 0;
-    if (ac && !bc) return 1;
-    if (!ac && bc) return -1;
-    if (ac > bc) return 1;
-    if (ac < bc) return -1;
-  }
-};
-
-const sortedLocations = locations => [...locations].sort(nestedSort);
-
-const memoizedSortedLocations = weakMemoizeArray(sortedLocations);
+const getterHelper = ['',''];
 
 const proxyShallowEqual = (a, b, locations) => {
   DISABLE_ALL_PROXIES = true;
+  const differ = [];
+  differs = [];
   const ret = (() => {
-    differs = [];
-    let valuables = null;
-    const nestedLocations = memoizedSortedLocations(locations);
-    let lastEqualKey = '';
-    for (let i = 0; i < nestedLocations.length; ++i) {
-      const key = nestedLocations[i];
-      if (
-        lastEqualKey &&
-        key.length > lastEqualKey.length &&
-        key[lastEqualKey.length] === '.' &&
-        key.indexOf(lastEqualKey) === 0
-      ) {
-        continue;
+    const root = memoizedBuildTrie(locations);
+    const walk = (la, lb, node) => {
```

Comment:
We can define `walk` outside of `proxyShallowEqual` which gives me better benchmark result. @theKashey

Reply:
How do you deal with `differ.unshift(item);` in walk?

Reply:
It's a global variable. Actually I should also extract `run`, to keep function as simple as possible.